toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs

```diff
@@ -230,40 +254,35 @@ fn prompt_permissioned_candidates_update_io(choice: bool) -> MockIO {
+		prompt_ogmios_configuration_io(
+			&default_ogmios_service_config(),
+			&default_ogmios_service_config(),
 		),
+		prompt_with_default_and_save_to_existing_file(
+			CARDANO_PAYMENT_SIGNING_KEY_FILE,
+			Some("payment.skey"),
+			"payment.skey",
+		),
+		MockIO::file_read("payment.skey"),
 		MockIO::print(
 			"Permissioned candidates updated. The change will be effective in two main chain epochs.",
 		)])
 }
 
-fn update_permissioned_candidates_io() -> MockIO {
+fn upsert_permissioned_candidates_failed_io() -> MockIO {
 	MockIO::Group(vec![
-		establish_pc_contracts_cli_config_io(),
-		MockIO::file_read("partner-chains-cli-chain-config.json"),
-		MockIO::run_command(
-			"./pc-contracts-cli update-permissioned-candidates --remove-all-candidates --network testnet --add-candidate 020a1091341fe5664bfa1782d5e04779689068c916b04cb365ec3153755684d9a1:d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d:88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee --add-candidate 0390084fdbf27d2b79d26a4f13f0ccd982cb755a661969143c37cbc49ef5b91f27:8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48:d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69 --genesis-utxo 0000000000000000000000000000000000000000000000000000000000000000#0 --kupo-host localhost --kupo-port 1442  --ogmios-host localhost --ogmios-port 1337  --payment-signing-key-file payment.skey",
-			"{\"endpoint\":\"UpdatePermissionedCandidates\",\"transactionId\":\"bbb5ab1232fde32884678333b44aa4f92a2229d7ba7a65d9eae4cb8b8c87d735\"}"
+		prompt_ogmios_configuration_io(
+			&default_ogmios_service_config(),
+			&default_ogmios_service_config(),
 		),
-		MockIO::print(
-		"Permissioned candidates updated. The change will be effective in two main chain epochs.",
-	)])
-}
-
-fn update_permissioned_candidates_failed_io() -> MockIO {
-	MockIO::Group(vec![
-		establish_pc_contracts_cli_config_io(),
-		MockIO::file_read("partner-chains-cli-chain-config.json"),
-		MockIO::run_command_with_result(
-			"./pc-contracts-cli update-permissioned-candidates --remove-all-candidates --network testnet --add-candidate 020a1091341fe5664bfa1782d5e04779689068c916b04cb365ec3153755684d9a1:d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d:88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee --add-candidate 0390084fdbf27d2b79d26a4f13f0ccd982cb755a661969143c37cbc49ef5b91f27:8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48:d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69 --genesis-utxo 0000000000000000000000000000000000000000000000000000000000000000#0 --kupo-host localhost --kupo-port 1442  --ogmios-host localhost --ogmios-port 1337  --payment-signing-key-file payment.skey",
-			Err(anyhow::anyhow!("Something went wrong"))
+		prompt_with_default_and_save_to_existing_file(
+			CARDANO_PAYMENT_SIGNING_KEY_FILE,
+			Some("payment.skey"),
+			"payment.skey",
 		),
+		MockIO::file_read("payment.skey"),
```

Comment:
I think few lines below there is now unused `establish_pc_contracts_cli_config_io`, please remove it